core/model/modx/sources/modfilemediasource.class.php

```diff
@@ -873,7 +873,7 @@ public function uploadObjectsToContainer($container,array $objects = array()) {
             }
 
             $newPath = $this->fileHandler->sanitizePath($file['name']);
-            $newPath = $directory->getPath().$newPath;
+            $newPath = ltrim(strip_tags(preg_replace('/[\.]{2,}/', '', htmlspecialchars($directory->getPath().$newPath))));
```

Comment:
Why not move this in sanitizePath?

Reply:
I have fears to break other uses case fileHandler->sanitizePath.
This method might have been used before in external components and their processors

Reply:
O.k. Then we maybe have to get the reason why two dots in the filename create that issue. 

Some sanitize method on the filename removes too much elsewhere. It just should remove the possibility to ascend in the path and strips those two dots instead of slashes/backslashes surrounded by two dots.

Reply:
If then it's also allowing to delete a file with two or multiple dots, than I'm alright with this 🚀